API/trading_api/trading_api_adapter.py

```diff
@@ -0,0 +1,74 @@
+import requests
+from constants import key
+
+
+class Adapter:
+    def __init__(self, frequency, asset, number_of_entries):
+        self.frequency = frequency
+        self.asset = asset
+        self.number_of_entries = number_of_entries
+
+    def validate_frequency(self):
+        frequencies = {
+            "daily": "TIME_SERIES_DAILY",
+            "weekly": "TIME_SERIES_WEEKLY",
+            "monthly": "TIME_SERIES_MONTHLY"
+        }
+        if self.frequency not in frequencies:
+            raise Exception(
+                "Wrong frequency. Use 'daily', 'monthly' or 'weekly' ")
```

Comment:
Just a nitpick, it seems weird to have them sorted in a string in alphabetic order :D And maybe it could be more general to have here an f-string (`f"{frequencies}"`) to list all the available options. Otherwise the exception might get outdated if you would add some frequency (e.g., bi-weekly).

Reply:
Noted, fixed